src/test/java/com/VetApp/PurrgentCare/services/AccountServiceImplementationTest.java

```diff
@@ -51,127 +51,134 @@ public void setup() {
         // given
         final var fakeAccountId = fakeDataGenerator.generateRandomInteger();
-        final var fakeActive = fakeDataGenerator.generateRandomBoolean();
-        final var fakeDateCreated = fakeDataGenerator.generateRandomDate();
-        final var fakePetList = fakeDataGenerator.generateDefaultPetList();
-        final var fakePeopleList = fakeDataGenerator.generatePersonList(fakeDataGenerator.generateRandomInteger());
-        final var fakeAccount = fakeDataGenerator.generateAccount(fakeAccountId, fakeActive, fakeDateCreated, fakePetList, fakePeopleList);
+        final var fakeAccountResponse = new AccountResponse();
+        given(mockAccountRepository.findById(fakeAccountId)).willReturn(Optional.empty());
 
         // when
-        serviceUnderTest.addAccount(fakeAccount);
+        final var actual = serviceUnderTest.getAccount(fakeAccountId);
 
         // then
-        verify(mockAccountRepository, times(1)).save(fakeAccount);
+        then(actual).isInstanceOf(AccountResponse.class);
+        then(actual).usingRecursiveComparison().isEqualTo(fakeAccountResponse);
     }
 
     @Test
-    public void deleteAccount_whenValidInput() {
+    public void getAllAccounts_withValidInput_returnsAllAccounts() {
         // given
-        final var fakeAccountId = fakeDataGenerator.generateRandomInteger();
+        final var fakeAccounts = fakeDataGenerator.generateDefaultAccountList();
+        final var fakeAccountResponses = fakeDataGenerator.generateFakeAccountResponses(fakeAccounts.size());
+        given(mockAccountRepository.findAll())
+                .willReturn(fakeAccounts);
+        for (var i = 0; i < fakeAccounts.size(); i++) {
+            given(mockMapper.map(fakeAccounts.get(i), AccountResponse.class))
+                    .willReturn(fakeAccountResponses.get(i));
+        }
 
         // when
-        serviceUnderTest.deleteAccount(fakeAccountId);
+        final var actual = serviceUnderTest.getAllAccounts();
 
         // then
-        verify(mockAccountRepository, times(1)).deleteById(fakeAccountId);
-
+        then(actual).isEqualTo(fakeAccountResponses);
+        then(actual).hasSize(fakeAccountResponses.size());
     }
 
     @Test
-    public void getAllAccounts_withValidInput_returnsAllAccounts() {
+    public void getAllAccounts_whenNoAccounts_returnsEmptyList() {
         // given
-        final var countOfAccounts = fakeDataGenerator.generateRandomInteger();
-        final var expected = fakeDataGenerator.generateAccountList(countOfAccounts);
-        given(mockAccountRepository.findAll())
-                .willReturn(expected);
+        final var fakeAccounts = new ArrayList<Account>();
+        final var fakeAccountResponses = new ArrayList<AccountResponse>();
+        given(mockAccountRepository.findAll()).willReturn(fakeAccounts);
 
         // when
         final var actual = serviceUnderTest.getAllAccounts();
+        verify(mockMapper, never()).map(any(), eq(AccountResponse.class));
```

Comment:
this should be in the then portion b/c it's a validation step but love that it's included in general.

Reply:
moved